lib/common/util/util.js

```diff
@@ -29,14 +31,29 @@ export function versionsSorter (
   return 0
 }
 
+/**
+ * Obtains the displayed (localizable) text for the given feed retrieval method.
+ */
 export function retrievalMethodString (method: string): string {
+  // Get the retrieval method strings.
+  const messages = getComponentMessages('FeedSourceViewer')
+  const retrievalMethod = 'properties.retrievalMethod'
+
   switch (method) {
-    case 'MANUALLY_UPLOADED':
-      return 'Manually Uploaded'
-    case 'FETCHED_AUTOMATICALLY':
-      return 'Fetched Automatically'
-    case 'PRODUCED_IN_HOUSE':
-      return 'Produced In-house'
+    case RETRIEVAL_METHODS.MANUALLY_UPLOADED:
+      return messages(`${retrievalMethod}.manuallyUploaded`)
+    case RETRIEVAL_METHODS.FETCHED_AUTOMATICALLY:
+      return messages(`${retrievalMethod}.fetchedAutomatically`)
+    case RETRIEVAL_METHODS.PRODUCED_IN_HOUSE:
+      return messages(`${retrievalMethod}.producedInHouse`)
+    case RETRIEVAL_METHODS.PRODUCED_IN_HOUSE_GTFS_PLUS:
+      return messages(`${retrievalMethod}.producedInHouseGtfsPlus`)
+    case RETRIEVAL_METHODS.REGIONAL_MERGE:
+      return messages(`${retrievalMethod}.regionalMerge`)
+    case RETRIEVAL_METHODS.SERVICE_PERIOD_MERGE:
+      return messages(`${retrievalMethod}.servicePeriodMerge`)
+    case RETRIEVAL_METHODS.VERSION_CLONE:
+      return messages(`${retrievalMethod}.versionClone`)
```

Comment:
If we want to be dynamic with these message keys, we could just convert the snake case with camel case and join with `properties.retrievalMethod`. But if we're defining a case for each of these, why not just have each be a string literal? E.g., `return messages('properties.retrievalMethod.manuallyUploaded')`

Reply:
Thanks for the suggestion. Updated in https://github.com/ibi-group/datatools-ui/pull/692/commits/0bebaf0034ffa431535039c11648ceb651f220d2.

Reply:
Oops, looks like that broke things, @binh-dam-ibigroup.